feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -596,20 +600,14 @@ class BillingWrapper(
         purchases: List<Purchase>?
     ) {
         val notNullPurchasesList = purchases ?: emptyList()
-        if (billingResult.responseCode == BillingClient.BillingResponseCode.OK && notNullPurchasesList.isNotEmpty()) {
+        if (billingResult.responseCode == BillingClient.BillingResponseCode.OK) {
             val storeTransactions = mutableListOf<StoreTransaction>()
-
             notNullPurchasesList.forEach { purchase ->
-                getStoreTransaction(purchase) { storeTxn ->
-                    storeTransactions.add(storeTxn)
-                    if (storeTransactions.size == notNullPurchasesList.size) {
-                        purchasesUpdatedListener?.onPurchasesUpdated(storeTransactions)
-                    }
-                }
+                getStoreTransaction(
+                    purchase,
+                    { tx -> storeTransactions.add(tx) })
             }
-        } else if (billingResult.responseCode == BillingClient.BillingResponseCode.OK) {
-            // When doing a DEFERRED downgrade, the result is OK, but the list of purchases is null
-            purchasesUpdatedListener?.onPurchasesUpdated(emptyList())
+            purchasesUpdatedListener?.onPurchasesUpdated(storeTransactions)
```

Comment:
As far as I see, `queryPurchasesAsyncWithTracking` will get called if there is no `PurchaseContext` for the given product id (since `productType` is non-nullable).

https://github.com/RevenueCat/purchases-android/blob/dde26e0f96033aced7581d772654026eb0d8c554/feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt#L750C22-L765

`PurchaseContext` is created when calling `makePurchaseAsync`.

https://github.com/RevenueCat/purchases-android/blob/dde26e0f96033aced7581d772654026eb0d8c554/feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt#L258

Is there a case where the former will be called but not the latter?